### ISSUE TITLE
Unable to Install

### ISSUE THREAD

**Issue by Zachattack525** [Virtuoel/Pehkui#35]
Quite simply, whenever I attempt to install this mod, one of two things happens: it either crashes on startup, or it simply doesn't load the mod.

**Comment by Gaming32** [comment]
I had the same issue:
```
17:07:29.227
Mixin prepare failed preparing AbstractDonkeyEntityMixin in pehkui.mixins.json: org.spongepowered.asm.mixin.transformer.throwables.InvalidMixinException The specified mixin 'virtuoel.pehkui.mixin.AbstractDonkeyEntityMixin' was not found
org.spongepowered.asm.mixin.transformer.throwables.InvalidMixinException: The specified mixin 'virtuoel.pehkui.mixin.AbstractDonkeyEntityMixin' was not found
	at org.spongepowered.asm.mixin.transformer.MixinInfo.<init>(MixinInfo.java:855)
	at org.spongepowered.asm.mixin.transformer.MixinConfig.prepareMixins(MixinConfig.java:715)
	at org.spongepowered.asm.mixin.transformer.MixinConfig.prepare(MixinConfig.java:648)
	at org.spongepowered.asm.mixin.transformer.MixinProcessor.prepareConfigs(MixinProcessor.java:514)
	at org.spongepowered.asm.mixin.transformer.MixinProcessor.select(MixinProcessor.java:439)
	at org.spongepowered.asm.mixin.transformer.MixinProcessor.checkSelect(MixinProcessor.java:415)
	at org.spongepowered.asm.mixin.transformer.MixinProcessor.applyMixins(MixinProcessor.java:285)
	at org.spongepowered.asm.mixin.transformer.MixinTransformer.transformClass(MixinTransformer.java:208)
	at org.spongepowered.asm.mixin.transformer.MixinTransformer.transformClassBytes(MixinTransformer.java:178)
	at org.spongepowered.asm.mixin.transformer.FabricMixinTransformerProxy.transformClassBytes(FabricMixinTransformerProxy.java:23)
	at net.fabricmc.loader.launch.knot.KnotClassDelegate.getPostMixinClassByteArray(KnotClassDelegate.java:157)
	at net.fabricmc.loader.launch.knot.KnotClassLoader.loadClass(KnotClassLoader.java:150)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	at java.lang.Class.forName0(Native Method)
	at java.lang.Class.forName(Class.java:348)
	at net.fabricmc.loader.util.DefaultLanguageAdapter.create(DefaultLanguageAdapter.java:45)
	at net.fabricmc.loader.EntrypointStorage$NewEntry.create(EntrypointStorage.java:112)
	at net.fabricmc.loader.EntrypointStorage$NewEntry.getOrCreate(EntrypointStorage.java:99)
	at net.fabricmc.loader.EntrypointStorage.lambda$getEntrypointContainers$1(EntrypointStorage.java:184)
	at net.fabricmc.loader.EntrypointStorage$$Lambda$83/1438988851.get(Unknown Source)
	at net.fabricmc.loader.entrypoint.EntrypointContainerImpl.getEntrypoint(EntrypointContainerImpl.java:37)
	at net.fabricmc.loader.entrypoint.minecraft.hooks.EntrypointUtils.invoke0(EntrypointUtils.java:50)
	at net.fabricmc.loader.entrypoint.minecraft.hooks.EntrypointUtils.invoke(EntrypointUtils.java:36)
	at net.fabricmc.loader.launch.knot.Knot.init(Knot.java:137)
	at net.fabricmc.loader.launch.knot.KnotClient.main(KnotClient.java:27)
Caused by: java.lang.ClassNotFoundException: The specified mixin 'virtuoel.pehkui.mixin.AbstractDonkeyEntityMixin' was not found
	at org.spongepowered.asm.mixin.transformer.MixinInfo.loadMixinClass(MixinInfo.java:1297)
	at org.spongepowered.asm.mixin.transformer.MixinInfo.<init>(MixinInfo.java:848)
	... 24 more
17:07:29.279

Exception in thread "main" java.lang.RuntimeException: Could not execute entrypoint stage 'preLaunch' due to errors, provided by 'replaymod'!
	at net.fabricmc.loader.entrypoint.minecraft.hooks.EntrypointUtils.invoke0(EntrypointUtils.java:53)
	at net.fabricmc.loader.entrypoint.minecraft.hooks.EntrypointUtils.invoke(EntrypointUtils.java:36)
	at net.fabricmc.loader.launch.knot.Knot.init(Knot.java:137)
	at net.fabricmc.loader.launch.knot.KnotClient.main(KnotClient.java:27)
Caused by: org.spongepowered.asm.mixin.throwables.MixinApplyError: Mixin [pehkui.mixins.json:AbstractDonkeyEntityMixin] from phase [DEFAULT] in config [pehkui.mixins.json] FAILED during PREPARE
	at org.spongepowered.asm.mixin.transformer.MixinProcessor.handleMixinError(MixinProcessor.java:642)
	at org.spongepowered.asm.mixin.transformer.MixinProcessor.handleMixinPrepareError(MixinProcessor.java:590)
	at org.spongepowered.asm.mixin.transformer.MixinProcessor.prepareConfigs(MixinProcessor.java:517)
	at org.spongepowered.asm.mixin.transformer.MixinProcessor.select(MixinProcessor.java:439)
	at org.spongepowered.asm.mixin.transformer.MixinProcessor.checkSelect(MixinProcessor.java:415)
	at org.spongepowered.asm.mixin.transformer.MixinProcessor.applyMixins(MixinProcessor.java:285)
	at org.spongepowered.asm.mixin.transformer.MixinTransformer.transformClass(MixinTransformer.java:208)
	at org.spongepowered.asm.mixin.transformer.MixinTransformer.transformClassBytes(MixinTransformer.java:178)
	at org.spongepowered.asm.mixin.transformer.FabricMixinTransformerProxy.transformClassBytes(FabricMixinTransformerProxy.java:23)
	at net.fabricmc.loader.launch.knot.KnotClassDelegate.getPostMixinClassByteArray(KnotClassDelegate.java:157)
	at net.fabricmc.loader.launch.knot.KnotClassLoader.loadClass(KnotClassLoader.java:150)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	at java.lang.Class.forName0(Native Method)
	at java.lang.Class.forName(Class.java:348)
	at net.fabricmc.loader.util.DefaultLanguageAdapter.create(DefaultLanguageAdapter.java:45)
	at net.fabricmc.loader.EntrypointStorage$NewEntry.create(EntrypointStorage.java:112)
	at net.fabricmc.loader.EntrypointStorage$NewEntry.getOrCreate(EntrypointStorage.java:99)
	at net.fabricmc.loader.EntrypointStorage.lambda$getEntrypointContainers$1(EntrypointStorage.java:184)
	at net.fabricmc.loader.EntrypointStorage$$Lambda$83/1438988851.get(Unknown Source)
	at net.fabricmc.loader.entrypoint.EntrypointContainerImpl.getEntrypoint(EntrypointContainerImpl.java:37)
	at net.fabricmc.loader.entrypoint.minecraft.hooks.EntrypointUtils.invoke0(EntrypointUtils.java:50)
	... 3 more
Caused by: org.spongepowered.asm.mixin.transformer.throwables.InvalidMixinException: The specified mixin 'virtuoel.pehkui.mixin.AbstractDonkeyEntityMixin' was not found
	at org.spongepowered.asm.mixin.transformer.MixinInfo.<init>(MixinInfo.java:855)
	at org.spongepowered.asm.mixin.transformer.MixinConfig.prepareMixins(MixinConfig.java:715)
	at org.spongepowered.asm.mixin.transformer.MixinConfig.prepare(MixinConfig.java:648)
	at org.spongepowered.asm.mixin.transformer.MixinProcessor.prepareConfigs(MixinProcessor.java:514)
	... 21 more
Caused by: java.lang.ClassNotFoundException: The specified mixin 'virtuoel.pehkui.mixin.AbstractDonkeyEntityMixin' was not found
	at org.spongepowered.asm.mixin.transformer.MixinInfo.loadMixinClass(MixinInfo.java:1297)
	at org.spongepowered.asm.mixin.transformer.MixinInfo.<init>(MixinInfo.java:848)
	... 24 more
17:07:29.279
Process crashed with exit code 1
```
is my console output.

**Comment by Virtuoel** [comment]
I am unable to reproduce this issue. Can you try again with Pehkui 1.8.0? If you still have the issue, make sure to also include the exact versions of Minecraft, Fabric Loader, Fabric API, and versions of any other mods that you're using.

**Comment by LordMZTE** [comment]
I am also getting this issue. [This is my log](https://paste.ee/p/cgIRH) all versions should be included.
Looks like this is caused from Downloading the newest version from modrinth, because it seems like someone accidentally uploaded the sources instead of the correct mod jar file.

**Comment by Virtuoel** [comment]
From that log you seem to be using the `-sources` jar. Make sure you download the non `-sources` jar. If you're getting the jar from Modrinth, you may need to pick the other jar from the specific version instead of just using the download button, as they appear to have a bug currently where I cannot pick which file gets downloaded by default.

**Comment by LordMZTE** [comment]
Well, as I said, I know those are the sources. I didn't know the right jar was on modrinth tho. But why are the sources uploaded to modrinth anyways?

**Comment by Virtuoel** [comment]
The sources are uploaded to Modrinth (also to CurseForge and GitHub releases) mainly for developers who want to make a mod that uses Pehkui and use Modrinth as one of their maven repositories for where they get their dependency mods and sources from.

**Comment by TimeBends** [comment]
I also make the same problem, where the mod simply doesn't load, I only have the latest version of  Immersive portals, world edit, and the fabric api, I've tried all the 1.16.5 versions of Pehkui, but still doesn't work, I even removed the mods, still the same issue.

**Comment by Gaming32** [comment]
This was fixed for me after installing the normal JAR instead of the
`-sources` JAR.

**Comment by Virtuoel** [comment]
Should now be resolved in 1.9.0.
